runtime/Cpp/runtime/src/support/StringUtils.h

```diff
@@ -21,11 +21,10 @@ namespace antlrcpp {
   template<typename T>
   inline std::string utf32_to_utf8(T const& data)
   {
-    // Don't make the converter static or we have to serialize access to it.
-    thread_local UTF32Converter converter;
+    UTF32Converter converter;
```

Comment:
Hi Mike, 

there is no way that removing of thread_local can lead do crashes. I guess it crashed, when "static" was used. The relevant CL was "bda96a0081c06126b55b1569cc1586ffd3cb3ce8", saying "getText performance improvement by making UTF32Converter thread_local instead of local.", so the author claims a performance improvement by using "thread_local".

Okay, I will revert it then ... although thread_local is a heavy weapon, not supported on some platforms, i.e. AIX xlcclang compiler on AIX 7.2, which otherwise support C++14. I was trying to compile Antlr 4 there ;-).

Best regards,
Kilian.

Reply:
Ah yes, you are right it was static. But you should not forget that with that local var the utf converter (which is pretty heavy) is instantiated for every `getText()` call, which is often used for predicates, slowing down the parser significantly. So yes, please revert that change. Thanks.